modules/persistence/src/services/metadata/repos_branches/index.ts

```diff
@@ -80,20 +114,20 @@ export const getRepoBranchesEntry = async (project: project, branch = ''): Promi
   // get from DB if not cached
   try {
     const db = await pool();
-    const query = {
-      project,
-    };
+    const matchCondition = { project };
     if (branch) {
-      query['branches'] = {
-        $elemMatch: { gitBranchName: branch },
-      };
+      matchCondition['branches'] = { $elemMatch: { gitBranchName: branch } };
     }
-    const res = (await db.collection('repos_branches').findOne(query)) as unknown as ReposBranchesDocument;
+    const aggregationPipeline = getAggregationPipeline(matchCondition);
+
+    const cursor = db.collection('docsets').aggregate(aggregationPipeline);
+    const res = (await cursor.toArray()) as unknown as ReposBranchesDocument[];
+
     // if not already set, set cache value for repo_branches
     if (!internals[project]) {
-      internals[project] = res;
+      internals[project] = res[0];
```

Comment:
Probably not relevant for the current state of our data, but I'm curious: since the `matchCondition` only expects a project and a branch, is it ever possible for 2 docs repos in the same docset with the same project and branch to both be in `res`? i.e. is the data we want always guaranteed to be at `res[0]`?

Reply:
This is interesting. I could see that. I could always add checking for the repo that is "prodDeployable" however I believe there are instances where we actually want to use docs-mongodb-internal. So, that kind of throws a wrench in it. 
This is the structure of the query that was findOne before, so I am now wondering if this needs to be a larger discussion about how we specify in these queries.

Reply:
Gotcha, yeah, I think it might be a bit ambiguous since the previous queries didn't really take into account different repos with the same project either.

I'm okay with things as-is, or with the extra check for `prodDeployable` as long as we're getting back the intended repo info (since associated products seem to mostly work with prod deploy data). Seems like there's pretty minimal risk with keeping things like this for now anyways, so I don't think this should be a blocker

Reply:
Agreed. And @seungpark does as well. I should have tagged this here.
For posterity, [ticket](https://jira.mongodb.org/browse/DOP-4044) created to address this.